### PR TITLE
[Easy] - add fantom prices & few tokens - Ready For Review 

### PR DESCRIPTION
**Pull request by henrystats**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by henrystats** [comment]
@jeff-dude is it possible to use a different symbol for a token than what's on coinpaprika? 

i'm getting error messages in the price automation check. - <https://github.com/duneanalytics/spellbook/actions/runs/3746190060/jobs/6361309490>

I also imported some tokens on ethereum and changed the contract addresses/decimals to the one on fantom.

Only bridged/major tokens like (WETH, WBTC, WFTM, etc)... It's returning that the contract addresses aren't listed in coinpaprika's contract.

also there's a blockchain naming error.

File: models/prices/prices_tokens.sql [hunk 1]
```diff
@@ -1756,5 +1757,56 @@ VALUES
     ("usdt-tether", "solana", "USDT", "Es9vMFrzaCERmJfrF4H2FYD4KCoNkY11McCe8BenwNYB", 6),
     ("drip-drip-network", "bnb", "DRIP", "0x20f663CEa80FaCE82ACDFA3aAE6862d246cE0333", 18),
     ("fur-furio", "bnb", "$FUR", "0x48378891d6E459ca9a56B88b406E8F4eAB2e39bF", 18),
-    ("pstn-piston", "bnb", "PSTN", "0xBfACD29427fF376FF3BC22dfFB29866277cA5Fb4", 18)
+    ("pstn-piston", "bnb", "PSTN", "0xBfACD29427fF376FF3BC22dfFB29866277cA5Fb4", 18),
+
+    ("weth-weth", "fantom", "WETH", "0x74b23882a30290451A17c44f4F05243b6b58C76d", 18), 
+    ("wbtc-wrapped-bitcoin", "fantom", "WBTC", "0x321162Cd933E2Be498Cd2267a90534A804051b11", 8),
+    ("ftm-fantom", "fantom", "WFTM", "0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83", 18),
```
Comment (Hosuke) [inline models/prices/prices_tokens.sql:1764]: `ERROR:root:ERROR: ftm-fantom Provided symbol: WFTM does not match CoinPaprika source: FTM`
- SCAN: https://ftmscan.com/token/0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83   (Wrapped Fantom)

File: models/prices/prices_tokens.sql [hunk 2]
```diff
@@ -1756,5 +1757,56 @@ VALUES
...
+    ("aave-new", "fantom", "AAVE", "0x6a07A792ab2965C72a5B8088d3a069A7aC3a993B", 18),
```
Comment (Hosuke) [inline models/prices/prices_tokens.sql:1765]: `ERROR:root:WARN: aave-new Provided contract address: 0x6a07A792ab2965C72a5B8088d3a069A7aC3a993B not in CoinPaprika contracts list ['0x7fc66500c84a76ad7e9c93437bfc5ac33e2ddae9']. (Not uncommon! share block explorer link to confirm contract)`
- API: https://api.coinpaprika.com/v1/coins/aave-new
- SCAN: https://ftmscan.com/token/0x6a07A792ab2965C72a5B8088d3a069A7aC3a993B

File: models/prices/prices_tokens.sql [hunk 3]
```diff
@@ -1756,5 +1757,56 @@ VALUES
...
     ("drip-drip-network", "bnb", "DRIP", "0x20f663CEa80FaCE82ACDFA3aAE6862d246cE0333", 18),
     ("fur-furio", "bnb", "$FUR", "0x48378891d6E459ca9a56B88b406E8F4eAB2e39bF", 18),
-    ("pstn-piston", "bnb", "PSTN", "0xBfACD29427fF376FF3BC22dfFB29866277cA5Fb4", 18)
+    ("pstn-piston", "bnb", "PSTN", "0xBfACD29427fF376FF3BC22dfFB29866277cA5Fb4", 18),
+
+    ("weth-weth", "fantom", "WETH", "0x74b23882a30290451A17c44f4F05243b6b58C76d", 18), 
+    ("wbtc-wrapped-bitcoin", "fantom", "WBTC", "0x321162Cd933E2Be498Cd2267a90534A804051b11", 8),
+    ("ftm-fantom", "fantom", "WFTM", "0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83", 18),
+    ("aave-new", "fantom", "AAVE", "0x6a07A792ab2965C72a5B8088d3a069A7aC3a993B", 18), 
+    ("avax-avalanche", "fantom", "AVAX", "0x511D35c52a3C244E7b8bd92c0C297755FbD89212", 18),
```
Comment (Hosuke) [inline models/prices/prices_tokens.sql:1766]: `ERROR:root:WARN: avax-avalanche Provided contract address: 0x511D35c52a3C244E7b8bd92c0C297755FbD89212 not in CoinPaprika contracts list ['api response missing contracts field']. (Not uncommon! share block explorer link to confirm contract)`
- API: https://api.coinpaprika.com/v1/coins/avax-avalanche
- SCAN: https://ftmscan.com/token/0x511D35c52a3C244E7b8bd92c0C297755FbD89212

File: models/prices/prices_tokens.sql [hunk 4]
```diff
@@ -1756,5 +1757,56 @@ VALUES
...
     ("fur-furio", "bnb", "$FUR", "0x48378891d6E459ca9a56B88b406E8F4eAB2e39bF", 18),
-    ("pstn-piston", "bnb", "PSTN", "0xBfACD29427fF376FF3BC22dfFB29866277cA5Fb4", 18)
+    ("pstn-piston", "bnb", "PSTN", "0xBfACD29427fF376FF3BC22dfFB29866277cA5Fb4", 18),
+
+    ("weth-weth", "fantom", "WETH", "0x74b23882a30290451A17c44f4F05243b6b58C76d", 18), 
+    ("wbtc-wrapped-bitcoin", "fantom", "WBTC", "0x321162Cd933E2Be498Cd2267a90534A804051b11", 8),
+    ("ftm-fantom", "fantom", "WFTM", "0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83", 18),
+    ("aave-new", "fantom", "AAVE", "0x6a07A792ab2965C72a5B8088d3a069A7aC3a993B", 18), 
+    ("avax-avalanche", "fantom", "AVAX", "0x511D35c52a3C244E7b8bd92c0C297755FbD89212", 18), 
+    ("crv-curve-dao-token", "fantom", "CRV", "0x1E4F97b9f9F913c46F1632781732927B9019C68b", 18),
```
Comment (Hosuke) [inline models/prices/prices_tokens.sql:1767]: `ERROR:root:WARN: crv-curve-dao-token Provided contract address: 0x1E4F97b9f9F913c46F1632781732927B9019C68b not in CoinPaprika contracts list ['0xd533a949740bb3306d119cc777fa900ba034cd52']. (Not uncommon! share block explorer link to confirm contract)`
- API: https://api.coinpaprika.com/v1/coins/crv-curve-dao-token
- SCAN: https://ftmscan.com/token/0x1E4F97b9f9F913c46F1632781732927B9019C68b

File: models/prices/prices_tokens.sql [hunk 5]
```diff
@@ -1756,5 +1757,56 @@ VALUES
...
+    ("dai-dai", "fantom", "DAI", "0x8D11eC38a3EB5E956B052f67Da8Bdc9bef8Abf3E", 18),
```
Comment (Hosuke) [inline models/prices/prices_tokens.sql:1768]: `ERROR:root:WARN: dai-dai Provided contract address: 0x8D11eC38a3EB5E956B052f67Da8Bdc9bef8Abf3E not in CoinPaprika contracts list ['0x6b175474e89094c44da98b954eedeac495271d0f']. (Not uncommon! share block explorer link to confirm contract)`
- API: https://api.coinpaprika.com/v1/coins/dai-dai
- SCAN: https://ftmscan.com/token/0x8d11ec38a3eb5e956b052f67da8bdc9bef8abf3e

File: models/prices/prices_tokens.sql [hunk 6]
```diff
@@ -1756,5 +1757,56 @@ VALUES
...
+    ("frax-frax", "fantom", "FRAX", "0xdc301622e621166BD8E82f2cA0A26c13Ad0BE355", 18),
```
Comment (Hosuke) [inline models/prices/prices_tokens.sql:1769]: `ERROR:root:WARN: frax-frax Provided contract address: 0xdc301622e621166BD8E82f2cA0A26c13Ad0BE355 not in CoinPaprika contracts list ['0x853d955acef822db058eb8505911ed77f175b99e', '0x29ced01c447166958605519f10dcf8b0255fb379']. (Not uncommon! share block explorer link to confirm contract)`
- API: https://api.coinpaprika.com/v1/coins/frax-frax
- SCAN: https://ftmscan.com/token/0xdc301622e621166BD8E82f2cA0A26c13Ad0BE355

File: models/prices/prices_tokens.sql [hunk 7]
```diff
@@ -1756,5 +1757,56 @@ VALUES
...
+    ("renbtc-renbtc", "fantom", "renBTC", "0xDBf31dF14B66535aF65AaC99C32e9eA844e14501", 8),
```
Comment (Hosuke) [inline models/prices/prices_tokens.sql:1770]: `ERROR:root:WARN: renbtc-renbtc Provided contract address: 0xDBf31dF14B66535aF65AaC99C32e9eA844e14501 not in CoinPaprika contracts list ['0xeb4c2781e4eba804ce9a9803c67d0893436bb27d']. (Not uncommon! share block explorer link to confirm contract)`
- API: https://api.coinpaprika.com/v1/coins/renbtc-renbtc
- SCAN: https://ftmscan.com/token/0xDBf31dF14B66535aF65AaC99C32e9eA844e14501

File: models/prices/prices_tokens.sql [hunk 8]
```diff
@@ -1756,5 +1757,56 @@ VALUES
...
+    ("usdc-usd-coin", "fantom", "USDC", "0x04068DA6C83AFCFA0e13ba15A6696662335D5B75", 6),
```
Comment (Hosuke) [inline models/prices/prices_tokens.sql:1771]: `ERROR:root:WARN: usdc-usd-coin Provided contract address: 0x04068DA6C83AFCFA0e13ba15A6696662335D5B75 not in CoinPaprika contracts list ['0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48']. (Not uncommon! share block explorer link to confirm contract)`
- API: https://api.coinpaprika.com/v1/coins/usdc-usd-coin
- SCAN: https://ftmscan.com/token/0x04068DA6C83AFCFA0e13ba15A6696662335D5B75

File: models/prices/prices_tokens.sql [hunk 9]
```diff
@@ -1756,5 +1757,56 @@ VALUES
...
+    ("crv-curve-dao-token", "fantom", "CRV", "0x1E4F97b9f9F913c46F1632781732927B9019C68b", 18),
+    ("dai-dai", "fantom", "DAI", "0x8D11eC38a3EB5E956B052f67Da8Bdc9bef8Abf3E", 18),
+    ("frax-frax", "fantom", "FRAX", "0xdc301622e621166BD8E82f2cA0A26c13Ad0BE355", 18),
+    ("renbtc-renbtc", "fantom", "renBTC", "0xDBf31dF14B66535aF65AaC99C32e9eA844e14501", 8),
+    ("usdc-usd-coin", "fantom", "USDC", "0x04068DA6C83AFCFA0e13ba15A6696662335D5B75", 6), 
+    ("tusd-trueusd", "fantom", "TUSD", "0x9879aBDea01a879644185341F7aF7d8343556B7a", 18), 
+    ("fxs-frax-share", "fantom", "FXS", "0x7d016eec9c25232b01F23EF992D98ca97fc2AF5a", 18),
+    ("mcrt-magiccraft", "fantom", "MCRT", "0xE705aF5f63fcaBDCDF5016aA838EAaac35D12890", 9), 
+    ("tarot-tarot", "fantom", "TAROT", "0xC5e2B037D30a390e62180970B3aa4E91868764cD", 18),
+    ("brush-brush", "fantom", "BRUSH", "0x85dec8c4B2680793661bCA91a8F129607571863d", 18),
```
Comment (Hosuke) [inline models/prices/prices_tokens.sql:1776]: `ERROR:root:WARN: brush-brush Provided contract address: 0x85dec8c4B2680793661bCA91a8F129607571863d not in CoinPaprika contracts list ['api response missing contracts field']. (Not uncommon! share block explorer link to confirm contract)`
- API: https://api.coinpaprika.com/v1/coins/brush-brush
- SCAN: https://ftmscan.com/token/0x85dec8c4B2680793661bCA91a8F129607571863d

**Review by Hosuke — APPROVED** [review]
I have manually checked those tokens in [token-prices-check warning](https://github.com/duneanalytics/spellbook/actions/runs/3746298065/jobs/6361506975#step:6). 
Other tokens not mentioned in the check should have passed the test I think.

**Comment by henrystats** [comment]
Thank you very much ser @Hosuke

**Comment by jeff-dude** [comment]
@0xRobin do you mind taking a quick glance at prices test failure here? did adding fantom break the script somehow?

**Comment by 0xRobin** [comment]
@jeff-dude I pushed a quick fix for the token checker.. 
It was indeed the fantom token with a null contract address that broke the checker. 
The output should now be as expected.

**Comment by jeff-dude** [comment]
> @jeff-dude I pushed a quick fix for the token checker.. It was indeed the fantom token with a null contract address that broke the checker. The output should now be as expected.

thank you!
@henrystats do you want to quickly scan through the warnings / errors in the output and ensure all looks good? i know you guys did checks on most already, but just final prior to merge.
https://github.com/duneanalytics/spellbook/actions/runs/3894980606/jobs/6649706065#step:6:23

**Comment by henrystats** [comment]
yes ser @jeff-dude, I just went through all the warnings / errors again and everything looks good to me! Thank you very much!

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**